Packages/ClientRuntime/Sources/Middleware/PresignerShimHandler.swift

```diff
@@ -24,19 +24,18 @@ struct PresignerShim<OperationStackOutput: HttpResponseBinding,
     
     public func handle<H>(context: HttpContext,
                           input: SdkHttpRequestBuilder,
-                          next: H) -> Result<OperationOutput<OperationStackOutput>, MError>
+                          next: H) async throws -> OperationOutput<OperationStackOutput>
     where H: Handler,
-    Self.Context == H.Context,
-    Self.MInput == H.Input,
-    Self.MOutput == H.Output,
-    Self.MError == H.MiddlewareError {
-        handler(input)
-        let httpResponse = HttpResponse(body: .none, statusCode: .ok)
-        do {
-            let output: OperationStackOutput? = try OperationStackOutput(httpResponse: httpResponse, decoder: nil)
-            return .success(.init(httpResponse: httpResponse, output: output))
-        } catch {
-            return .failure(.unknown(ClientError.unknownError("PresignerShimHandler: This code should not execute")))
-        }
-    }
+          Self.Context == H.Context,
+          Self.MInput == H.Input,
+          Self.MOutput == H.Output {
+              handler(input)
+              let httpResponse = HttpResponse(body: .none, statusCode: .ok)
+              do {
+                  let output: OperationStackOutput? = try OperationStackOutput(httpResponse: httpResponse, decoder: nil)
+                  return .init(httpResponse: httpResponse, output: output)
+              } catch {
+                  throw SdkError<OperationStackError>.unknown(ClientError.unknownError("PresignerShimHandler: This code should not execute"))
+              }
```

Comment:
Relooking at this code, it seems like we could get rid of `OperationStackError`, and just change tihs code to something like:
```
                  let output: OperationStackOutput? = try! OperationStackOutput(httpResponse: httpResponse, decoder: nil)
                  return .init(httpResponse: httpResponse, output: output)
```
lets just keep this as you have it for now, and we can update this in the future.